components/camera/transformpipeline/classifier.go

```diff
@@ -0,0 +1,106 @@
+package transformpipeline
+
+import (
+	"context"
+	"image"
+
+	"github.com/edaniels/gostream"
+	"github.com/pkg/errors"
+	"go.opencensus.io/trace"
+
+	"go.viam.com/rdk/components/camera"
+	"go.viam.com/rdk/config"
+	"go.viam.com/rdk/rimage/transform"
+	"go.viam.com/rdk/robot"
+	"go.viam.com/rdk/services/vision"
+	rdkutils "go.viam.com/rdk/utils"
+	"go.viam.com/rdk/vision/classification"
+)
+
+// classifierAttrs is the attribute struct for classifiers.
+type classifierAttrs struct {
+	ClassifierName      string  `json:"classifier_name"`
+	ConfidenceThreshold float64 `json:"confidence_threshold"`
+	MaxClassifications  *uint32 `json:"max_classifications"`
+}
+
+// classifierSource takes an image from the camera, and overlays labels from the classifier.
+type classifierSource struct {
+	stream             gostream.VideoStream
+	classifierName     string
+	maxClassifications uint32
+	confFilter         classification.Postprocessor
+	r                  robot.Robot
+}
+
+func newClassificationsTransform(
+	ctx context.Context,
+	source gostream.VideoSource, r robot.Robot, am config.AttributeMap,
+) (gostream.VideoSource, camera.ImageType, error) {
+	conf, err := config.TransformAttributeMapToStruct(&(classifierAttrs{}), am)
+	if err != nil {
+		return nil, camera.UnspecifiedStream, err
+	}
+	attrs, ok := conf.(*classifierAttrs)
+	if !ok {
+		return nil, camera.UnspecifiedStream, rdkutils.NewUnexpectedTypeError(attrs, conf)
+	}
+
+	props, err := propsFromVideoSource(ctx, source)
+	if err != nil {
+		return nil, camera.UnspecifiedStream, err
+	}
+	var cameraModel transform.PinholeCameraModel
+	cameraModel.PinholeCameraIntrinsics = props.IntrinsicParams
+
+	if props.DistortionParams != nil {
+		cameraModel.Distortion = props.DistortionParams
+	}
+	confFilter := classification.NewScoreFilter(attrs.ConfidenceThreshold)
+	var maxClassifications uint32 = 1
+	if attrs.MaxClassifications != nil {
+		maxClassifications = *attrs.MaxClassifications
+	}
```

Comment:
Maybe also make a && condition that it not equal 0, too.

Reply:
Sounds good, I intentionally made 0 configurable since I thought it might be an easy way for the user to "turn off" this feature, however we'll probably have a way to toggle that in the UI anyways so this probably causes more harm than good. Changed this.